eth/tracers/api_blocktrace.go

```diff
@@ -109,5 +110,18 @@ func (api *API) createTraceEnvAndGetBlockTrace(ctx context.Context, config *Trac
 	}
 
 	chaindb := api.backend.ChainDb()
-	return api.scrollTracerWrapper.CreateTraceEnvAndGetBlockTrace(api.backend.ChainConfig(), api.chainContext(ctx), api.backend.Engine(), chaindb, statedb, parent, block, true)
+	// create a copy of api.backend.ChainConfig to modify
+	chainConfig := new(params.ChainConfig)
+	*chainConfig = *api.backend.ChainConfig()
+	if config != nil && config.Overrides != nil {
+		// the merge.Merge seems not work well
+		// mergo.Merge(chainConfig, config.Overrides, mergo.WithOverride)
+		if curie := config.Overrides.CurieBlock; curie != nil {
+			chainConfig.CurieBlock = curie
+			misc.ApplyCurieHardFork(statedb)
+			statedb.Commit(true)
```

Comment:
i am not fully sure whether i should use `statedb.Commit(true)` or `statedb.IntermediateRoot(true)`.   Reviewers can have a check here.

Reply:
checked that `Commit` commits everything, including the following:
```
s.IntermediateRoot(deleteEmptyObjects)
```
pls also have a check.

Reply:
IIUC we can do `IntermediateRoot` per tx and `Commit` per block.   doing `Commit` looks safe here

Reply:
IntermediateRoot makes more sense.   For tracing, we don't need to commit trie updates to trie.Database.